server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -316,6 +314,11 @@ func mirrorToCache(ctx context.Context, bsClient bspb.ByteStreamClient, remoteIn
 		if expectedChecksum != "" && checksumDigestRN.GetDigest().GetHash() != expectedChecksum {
 			return nil, status.InvalidArgumentErrorf("response body checksum for %q was %q but wanted %q", uri, checksumDigestRN.GetDigest().Hash, expectedChecksum)
 		}
+		// for next request
+		_, err = cachetools.UploadFile(ctx, bsClient, remoteInstanceName, checksumFunc, tmpFilePath)
+		if err != nil {
+			return nil, status.UnavailableErrorf("failed to add object to cache: %s", err)
+		}
```

Comment:
Nit: the comment in this section should be updated.

Also, we do need a new test in `fetch_server_test.go` to properly validate this new logic.